### PR TITLE
feat: add changes api

### PR DESCRIPTION
**Pull request by elrrrrrrr**
Added a new api: `/-/all/changes`,
returns change records through the `Tag` table,
regardless of whether the package is private or public.

This api is mainly used for other registries for synchronization.

### CODE REVIEW

File: services/package.js [hunk 1]
```diff
@@ -198,6 +198,33 @@ exports.listPublicModuleNamesByUser = function* (username) {
   return names;
 };
 
+exports.listTagSince = function* listTagSince(start, limit, cursorId) {
+  if (!(start instanceof Date)) {
```
Comment (gemwuu) [inline services/package.js:202]: use `util.types.isDate(start)`

File: controllers/registry/package/changes.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+var packageService = require('../../../services/package');
+
+// GET /-/_changes?since={timestamp}&limit={number}&cursorId={number}
+// List packages names since the timestamp
+// Similar with https://replicate.npmjs.com/_changes?since=7139538
+module.exports = function* listSince() {
+  var query = this.query;
+  var since = query.since || '0';
+  var limit = Number(query.limit);
```
Comment (gemwuu) [inline controllers/registry/package/changes.js:11]: if `limit === 0`, no tag will be listed.

File: services/package.js (hunk 1 above)
Comment (elrrrrrrr) [reply services/package.js]: 👌🏻

File: controllers/registry/package/changes.js (hunk 2 above)
Comment (elrrrrrrr) [reply controllers/registry/package/changes.js:11]: If the `limit` is specified directly,
It should be the expected.

**Comment by killagu** [comment]
Changes should include package version release, tag modify, maintainer add/remove, unpublish.

**Comment by elrrrrrrr** [comment]
> Changes should include package version release, tag modify, maintainer add/remove, unpublish.

I have added some event types:  
 * `PACKAGE_TAG_ADDED` 
 * `PACKAGE_VERSION_ADDED` 
 * `PACKAGE_UNPUBLISHED`
 *  `PACKAGE_VERSION_BLOCKED`

But we only have the **updated** data.
So we can't  calculate some event message, such as `PACKAGE_TAG_CHANGED` `PACKAGE_MAINTAINER_ADDED` `PACKAGE_MAINTAINER_CHANGED`.

For the same reason, we can't calculate a **reliable** seqId.
let's use gmt_modified field to instead of .

**Review by killagu — APPROVED** [review]
LGTM

**Comment by killagu** [comment]
+ cnpmjs.org@3.0.0-rc.62

**Pull request merged**